src/externals/pio1/pio/piodarray.F90.in

```diff
@@ -655,13 +660,15 @@ contains
 #endif
        if(file%iotype==pio_iotype_pnetcdf) then
           call add_data_to_buffer(File, IOBUF, request)
-       else if(Userearranger) then
-          deallocate(iobuf)
+          is_iobuf_owned = .false.
        end if
 #ifdef TIMING
        call t_stopf("PIO:post_pio_write_nf")
 #endif
     end if
+    if(associated(IOBUF) .and. is_iobuf_owned) then
```

Comment:
why do you need the additional logical is_iobuf_owned?  associated should be enough.

Reply:
add_data_to_buffer takes ownership of IOBUF, I need to ensure that case didn't occur.